### ISSUE TITLE
Jellyfin reports "Object reference not set to an instance of an object." when browsing the movies list from VLC via DLNA

### ISSUE THREAD

**Issue by tussock** [jellyfin/jellyfin#10364]
### Please describe your bug

I have jellyfin running as a container (App) under Truenas scale.
I have enabled DLNA in Jellyfin.
DLNA sort of works - I can browse Movies/Collections, Movies/Genres, Music, but
other folders return nothing. Examples: Movies/Movies, Movies/Favorites.

When I look in the Jellyfin logs after attempting to open Movies/Movies, I see the following error:
```
[2023-10-08 11:44:31.896 +13:00] [ERR] [78] Emby.Dlna.ContentDirectory.ContentDirectoryService: Error processing control request
System.NullReferenceException: Object reference not set to an instance of an object.
   at MediaBrowser.Model.Dlna.StreamBuilder.BuildVideoItem(MediaSourceInfo item, VideoOptions options)
   at MediaBrowser.Model.Dlna.StreamBuilder.BuildVideoItem(VideoOptions options)
   at Emby.Dlna.Didl.DidlBuilder.AddVideoResource(XmlWriter writer, BaseItem video, String deviceId, Filter filter, StreamInfo streamInfo)
   at Emby.Dlna.Didl.DidlBuilder.WriteItemElement(XmlWriter writer, BaseItem item, User user, BaseItem context, Nullable`1 contextStubType, String deviceId, Filter filter, StreamInfo streamInfo)
   at Emby.Dlna.ContentDirectory.ControlHandler.HandleBrowse(XmlWriter xmlWriter, IReadOnlyDictionary`2 sparams, String deviceId)
   at Emby.Dlna.ContentDirectory.ControlHandler.WriteResult(String methodName, IReadOnlyDictionary`2 methodParams, XmlWriter xmlWriter)
   at Emby.Dlna.Service.BaseControlHandler.CreateControlResponse(ControlRequestInfo requestInfo)
   at Emby.Dlna.Service.BaseControlHandler.ProcessControlRequestInternalAsync(ControlRequest request)
   at Emby.Dlna.Service.BaseControlHandler.ProcessControlRequestAsync(ControlRequest request)
```

I see that just before the error occurred, the line:
```
[2023-10-08 11:44:31.896 +13:00] [INF] [78] Emby.Dlna.ContentDirectory.ContentDirectoryService: StreamBuilder.BuildVideoItem( Profile="Generic Device", Path="/mnt/movies/HomeMovie.mkv", AudioStreamIndex=null, SubtitleStreamIndex=null ) => ( PlayMethod=DirectPlay, TranscodeReason=0 ) "media:/videos/dab445fa-e7e6-6038-e997-053e92a20661/stream.mkv?MediaSourceId=dab445fae7e66038e997053e92a20661&Static=true&VideoCodec=h264&AudioCodec=ac3&AudioStreamIndex=1&api_key=<token>&Tag=b8f06c143f9aff84463f5222c584e76c"
```
appeared. From the log, I can see it was working through all the files starting with "H". So I removed files before and after "HomeMovie.mkv" just to see if there was a corruption in one of the files causing the problem.  No luck... the error still occurs.

I have checked permissions - all files and folders are 0755.
I have set jellyfin to only have read access to all my media files. 

All files work and browsing works via the jellyfin app on iOS and AppleTV and the web browser. None of the files mentioned above fail. This is only a DLNA issue.

### Jellyfin Version

Other

### if other:

10.8.11

### Environment

```markdown
- OS: Truenas scale
- Linux Kernel: Linux truenas 6.1.50-production+truenas #2 SMP PREEMPT_DYNAMIC Sat Sep 16 01:23:34 UTC 2023 x86_64 GNU/Linux
- Virtualization: Kubernetes
- Clients: Any DLNA
- Browser: Safari (but note - the problem doesn't occur when suing the browser.
- FFmpeg Version:
- Playback Method: DLNA browsing
- Hardware Acceleration: 
- GPU Model:
- Plugins:
- Reverse Proxy:
- Base URL:
- Networking:
- Storage:
```

### Jellyfin logs

```shell
[2023-10-08 11:44:31.896 +13:00] [INF] [78] Emby.Dlna.ContentDirectory.ContentDirectoryService: StreamBuilder.BuildVideoItem( Profile="Generic Device", Path="/mnt/movies/HomeMovie.mkv", AudioStreamIndex=null, SubtitleStreamIndex=null ) => ( PlayMethod=DirectPlay, TranscodeReason=0 ) "media:/videos/dab445fa-e7e6-6038-e997-053e92a20661/stream.mkv?MediaSourceId=dab445fae7e66038e997053e92a20661&Static=true&VideoCodec=h264&AudioCodec=ac3&AudioStreamIndex=1&api_key=<token>&Tag=b8f06c143f9aff84463f5222c584e76c"
[2023-10-08 11:44:31.896 +13:00] [INF] [78] Emby.Dlna.ContentDirectory.ContentDirectoryService: StreamBuilder.BuildVideoItem( Profile="Generic Device", Path="/mnt/movies/HomeMovie.mkv", AudioStreamIndex=null, SubtitleStreamIndex=null ) => ( PlayMethod=DirectPlay, TranscodeReason=0 ) "media:/videos/dab445fa-e7e6-6038-e997-053e92a20661/stream.mkv?MediaSourceId=dab445fae7e66038e997053e92a20661&Static=true&VideoCodec=h264&AudioCodec=ac3&AudioStreamIndex=1&api_key=<token>&Tag=b8f06c143f9aff84463f5222c584e76c"
```

### Please attach any screenshots here

![Screenshot 2023-10-08 at 11 58 03 AM](https://github.com/jellyfin/jellyfin/assets/811473/26ac38e7-5c26-4c01-b325-73610b26ca2f)

Note that the collections list is populated.
The Movies list is not - even after I have double clicked on it.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by felix920506** [comment]
Closing this one as it is likely a duplicate to #8075

**Comment by klausmcm** [comment]
@tussock I came across a similar issue which I noticed with 10.8.* and also still present in 10.8.13

This is the same for me

> other folders return nothing. Examples: Movies/Movies

Have you found a fix for this issue? I'm currently using 10.7.7 which does not have this issue.

**Comment by tussock** [comment]
> Have you found a fix for this issue? I'm currently using 10.7.7 which does not have this issue.

@klausmcm  I have not found a fix. I'm on 10.8.13.
This has been closed as a duplicate:
    https://github.com/jellyfin/jellyfin-plugin-dlna/issues/22
and they mention that ISO files or file positions might be a cause. I don't have ISO files,
and I have tried setting file perms to 0766 (a bad idea because it enables write).
Neither worked.

**Comment by klausmcm** [comment]
@tussock Thanks for letting me know. I actually do have an iso in my folder of movies but I have not tried removing the file yet. I'll try again in the near future and report back on any other observations.
